merge.go

```diff
@@ -487,7 +487,12 @@ func mergeTermFreqNormLocs(fieldsMap map[string]uint16, term []byte, postItr *Po
 		newRoaring.Add(uint32(hitNewDocNum))
 
 		nextFreq := next.Frequency()
-		nextNorm := uint64(math.Float32bits(float32(next.Norm())))
+		var nextNorm uint64
+		if pi, ok := next.(postingsAdv); ok {
+			nextNorm = pi.NormValue()
+		} else {
+			return 0, 0, 0, nil, fmt.Errorf("internal norm read error")
```

Comment:
i would make the error message more clear, something like

`fmt.Errorf("unexpected posting type %T", next)`